checkmate/checker/url/blocklist.py

```diff
@@ -37,11 +38,15 @@ def check_url(self, url):
         """Test the URL and return a list of reasons it should be blocked.
 
         :param url: URL to test
+        :raise MalformedURL: If the URL cannot be parsed
         :return: An iterable of Reason objects (which may be empty)
         """
         self._refresh()
 
         domain = self._domain(url)
+        if not domain:
+            raise MalformedURL(f"The URL: '{url}' has no domain to check")
```

Comment:
It doesn't seem right to make this a malformed parameter error as this class doesn't really know about parameters.

Reply:
I guess the other option might be to just rename `BadURLParameter` to just `BadURL`

Reply:
I think the other one is a `BadParameter` in general (does it have the wrong name?) but it could be a subclass